### ISSUE TITLE
One in the chamber

### ISSUE THREAD

**Issue by mvm900** [CleverRaven/Cataclysm-DDA#26370]
Picking up a gun without a magazine practically means the thing is worthless, especially early game when you lack the storage capacity to haul the thing around until you do find a mag, if you ever do. That's not even mentioning finding magazines is tedious since the only regular places to find them are gun stores. This is not always a huge problem but annoying enough that a quality of life improvement like this wouldn't hurt. 

The ability to load a single round into a gun even if it lacks a magazine, with some kind of penalty to reload speed or even increased probability of it jamming would easily rectify this. Most firearms allow this to some degree as long as you have access to the chamber and can pull the bolt back, in the case of rifles, smgs and the like, or can pull the slide back, in the case of handguns. This obviously isn't the safest procedure but, in a survival situation, justified. 

Alternatively, just having guns spawn with mags already with them. There's... not really a good reason to have a gun without the magazine either in it, somewhere close by or otherwise in the home with it. I dunno if having every gun that doesn't spawn without a mag have one nearby would be easy, so just having them already in would probably be less of a hassle.

**Comment by kevingranade** [comment]
> Picking up a gun without a magazine practically means the thing is worthless, especially early game when you lack the storage capacity to haul the thing around until you do find a mag, if you ever do.

There's not really any problem with that.

> The ability to load a single round into a gun even if it lacks a magazine ... would easily rectify this.

Try it sometime, if you find a gun with no magazine, wish for a magazine, but any time you want to reload the gun, drop a round on the ground, then reload the gun and tell it to use that round.
My suspicion is that the gun is STILL going to be mostly worthless since it takes so long to reload and only fires a single round.

> Most firearms allow this

Needs a list of guns this will work with, I am 100% certain that this ranges from wildly impractical to actually impossible depending on the gun in question.  "Most" doesn't turn into "all" just because you want it to.

> There's... not really a good reason to have a gun without the magazine either in it, somewhere close by or otherwise in the home with it.

Magazines are much more portable than guns, you can only meaningfully use a certain number of guns of the same kind, and the number of magazines of the same you can meaningfully use is much higher.  Therefore when looting or scavenging, people are likely to take just upgrades or side-grades to the guns they already have, but as many magazines as they can carry.  This applies to homes, gun shops, police and soldier corpses, armories, etc.

**Comment by l29ah** [comment]
All the working guns were already taken and used to their full extent in the first hours of the cataclysm so you're left with unlucky ones who were mismanaged (zombies aren't good at firearm maintenance i suppose). It's a pity there are no shoddy magazines that can be easily craftable from scrap tho.

**Comment by l29ah** [comment]
And there are lots of basic guns that don't have any makeshift magazines for them at all, afair.

**Comment by Vzor-** [comment]
> Needs a list of guns this will work with, I am 100% certain that this ranges from wildly impractical to actually impossible depending on the gun in question. "Most" doesn't turn into "all" just because you want it to.

I personally don't know of a gun that can't have a round manually inserted into the chamber. Manually checking the chamber for a bullet is a necessary feature for the safe operation of a firearm. To check in the dark, you actually put your finger in the chamber and feel for a casing.  If you can get you finger in, you can put a new round in. Though don't do it irl as it can damage the extractor on certain guns, I think the M1 even has a chance to slamfire. This could also be a chance to implement a "and one in the chamber" feature, allowing you to store 1 more round in a gun than the nominal capacity. 

I'm not saying either of these features should or shouldn't be in the game, just that they would reflect reality. As for how hard it is, with many guns you would need to push a follower down or press a bolt/slide release, nothing too terrible. e.g. https://www.youtube.com/watch?v=rhBfI_rFfGY

**Comment by mvm900** [comment]
> All the working guns were already taken and used to their full extent in the first hours of the cataclysm so you're left with unlucky ones who were mismanaged (zombies aren't good at firearm maintenance i suppose). It's a pity there are no shoddy magazines that can be easily craftable from scrap tho.

What does working guns have to do with the magazine? I can pull the trigger until it's empty, yes. But that doesn't mean the magazine disappears.

**Comment by mvm900** [comment]
> There's not really any problem with that.
It happens to me occasionally. Wouldn't hurt. 

> Try it sometime, if you find a gun with no magazine, wish for a magazine, but any time you want to reload the gun, drop a round on the ground, then reload the gun and tell it to use that round.
> My suspicion is that the gun is STILL going to be mostly worthless since it takes so long to reload and only fires a single round.
One round every ten minutes is superior to zero rounds ever. 

> Needs a list of guns this will work with, I am 100% certain that this ranges from wildly impractical to actually impossible depending on the gun in question. "Most" doesn't turn into "all" just because you want it to.
I don't imagine you have much firearm experience. On the contrary, you'd need to specifically look for weapons where this _doesn't_ work. The only example I can think of now is like, the G11 and weapons with caseless ammunition like it. Because all weapons have to eject a casing if they exist, you can potentially insert a bullet. Open bolt or closed bolt. 

> Magazines are much more portable than guns, you can only meaningfully use a certain number of guns of the same kind, and the number of magazines of the same you can meaningfully use is much higher. Therefore when looting or scavenging, people are likely to take just upgrades or side-grades to the guns they already have, but as many magazines as they can carry. This applies to homes, gun shops, police and soldier corpses, armories, etc.
This doesn't make much sense for an armory or gun store that's been locked up and the apocalypse is incredibly recent. Sure, a home maybe. But unless someone broke into one of these more secure locations and.. managed to take a mag and then leave and lock the door behind them while leaving all the other goodies, well, they must be some well off fellows. Also a gun is always useful provided you can find ammo and have the space for it. Either as a backup, one you can hide around your vehicle or whatever. Preppers often have more than one weapon for a reason. Not even mentioning that if it's a weapon of a different caliber you may not have ammo for it, but may know where to find some and the magazine won't be very useful to you if you don't have, you know, a weapon to shoot it from.

**Comment by kevingranade** [comment]
> It happens to me occasionally. Wouldn't hurt.

I'm not saying it doesn't happen, I'm saying the fact that it happens isn't a problem for the game, bad things happening to players aren't necessarally bugs.

> I don't imagine you have much firearm experience.

This kind of comment is unnecessary and you need to not make it, ok?

> On the contrary, you'd need to specifically look for weapons where this doesn't work.

Any weapon with a magazine disconnect (sometimes called an interlock) may allow you to physically insert a round into the chamber, but will not fire it without a magazine inserted. Police weapons in particular have a disproportionate preference for this feature.

> This doesn't make much sense for an armory or gun store that's been locked up and the apocalypse is incredibly recent.

Yes I'm talking about looted areas, which do exist, which refutes, "There's... not really a good reason to have a gun without the magazine either in it, somewhere close by or otherwise in the home with it.", there are a number of situations where this would happen, so "always spawn guns with matching magazines" is a non-starter.

**Comment by NotFuji** [comment]
Did some research, the only automatic in the current gun list that universally has a magazine disconnect is the FN Five-Seven.  All others only include one by special order or to comply with local laws.

**Comment by mvm900** [comment]
I don't seem to be able to quote on mobile, so forgive me. 

I never said it was a bug. I'm aware it's not possible and to some degree it's 'intended'; I believe it's a fault in game design if magazines are implemented and they're so rare despite them being so overly produced and cheap and you can't use the gun without them. 

Magazine disconnects are really rare to be standard on a weapon. The M9, popular police handgun, doesn't even have one standard. I'm also not aware that most get them with it since it's seen as such a downside and a huge possibility for errors during a gunfight. The inability to do so would really have to be one hell of an exception. 

The not looted areas exist with the same problem though, sometimes. And even then it assumes everyone is just out for a magazine which, for all intents and purposes, should be impeccably common really. 
And there exists similarly copious amounts of situations where that wouldn't happen as well.

**Comment by Profugo-Barbatus** [comment]
I'm inclined to agree with Kevin, from a gameplay perspective. Having a working gun before you get a vehicle is a lucky find, not something the player should be able to consistently do.

And there's no shortage of civilian grade weapons with internal magazines. I find bolt rifles and shotguns fairly often, both of which are more than serviceable in the early days. Revolvers aren't incredibly rare finds either, and those can pack quite the punch.

**Comment by keldoclock** [comment]
Sometimes it's easy to load just one round (most bolt action, shotgun, assault rifle, many pistols) and sometimes its a pain in the ass (M1 garand has an operating rod inside that will close the action if you touch it with your fingers while inserting). Magazine disconnects are not actually present on all guns in areas that require them because some are antiques, some simply predate the law, and they are a little piece of metal with a spring that can be removed with a screwdriver and nobody is checking for them. 

One thing this would add is the ability to interrupt a reload to aim and fire 1 shot, which is the real life reason many police officers and home owners remove their mag disconnect. 

It should be slow enough to chamber one that people won't want to do it in combat. Maybe crossbow reload amount of time.

**Comment by kevingranade** [comment]
See https://www.reddit.com/r/cataclysmdda/comments/9twobb/loading_guns_without_magazines/e924505

tl;dr add open-bolt guns and rotary guns to the list of, "can't chamber a round without a magazine".

**Comment by kevingranade** [comment]
This issue has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/let-us-load-single-rounds-into-guns-without-magazines/27495/2
